src/Crawler.py

```diff
@@ -1,14 +1,27 @@
-import typing
+import time
 from typing import List
+from concurrent.futures import ThreadPoolExecutor
+import threading
 
 from ScrapeResult import ScrapeResult
-from Database import Database
 from Scraper import Scraper
 
+'''
+Class for Crawler. Sets up threads and activates scrapers to crawl the internet, then write to file.
+'''
 class Crawler:
-    
+    starting_url = None
+    search_terms = None
+    max_num_threads = None
+    max_num_urls = None
```

Comment:
IIRC initialising to `None` is not necessary, doing `self.starting_url = <something>` in the `__init__` is enough.

Reply:
ok deleted